### ISSUE TITLE
[DEPRECATION] the default number of shards will change from [5] to [1] in 7.0.0

### ISSUE THREAD

**Issue by ambs** [elastic/elasticsearch-perl#166]
Hi
I am new to ES, so not sure what this means. Nevertheless, hope this information can be useful. Also, would be happy to know how to disable this issue.

````
[DEPRECATION] the default number of shards will change from [5] to [1] in 7.0.0; if you wish to continue using the default of [5] shards, you must manage this on the create index request or with an index template - In request: {body => ["{\"index\":{\"_id\":\"18_151\"}}","{\"tit...
`````

Thanks

**Comment by carlosfrodriguez** [comment]
Hi @ambs 

It seams that the module is outdated an does not support ES 7.0, I get deprecation errors and other failures all over the place.

The module hasn't been updated since years, so I hope that a new version will come soon.

**Comment by ambs** [comment]
It is curious that I only get this when storing some documents, as for others I get no warnings...

**Comment by carlosfrodriguez** [comment]
Well... provably it depends on your index settings.

**Comment by Grinnz** [comment]
You will want to pass `settings => { index => { number_of_shards => $num } }` in the body when creating the index. If the new default is fine then you can just ignore the deprecation message and it will change when you upgrade to 7.0.0.

**Comment by ezimuel** [comment]
We just released a dev version [7.30_1](https://github.com/elastic/elasticsearch-perl/releases/tag/version_7.30_1) supporting **Elasticsearch 7.3 APIs**. Please check this version and let me know, thanks!
